src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -84,112 +84,43 @@ private static unsafe int CompareOrdinalHelper(string strA, string strB)
                 "For performance reasons, callers of this method should " +
                 "check/short-circuit beforehand if the first char is the same.");
 
-            int length = Math.Min(strA.Length, strB.Length);
+            // Check if the second chars are different here
```

Comment:
Does this implementation effectively depend on this small wrapper to be inlined into the few callers for good perf / no regressions variety of inputs?

I am wondering whether it would be better to mark this small wrapper with AggressiveInlining and move the comparison of the first char (that's manually inlined today) into it as well.

Reply:
I have pushed a commit to address this feedback. It should address regression for small strings in non-tired case (proxy for AOT).